custom_components/mcp23017/__init__.py

```diff
@@ -134,18 +136,18 @@ async def async_unload_entry(hass, config_entry):
             if component.has_no_entities:
                 if component.is_alive():
                     await hass.async_add_executor_job(component.stop_polling)
-                hass.data[DOMAIN].pop(i2c_address)
+                hass.data[DOMAIN].pop(i2c_bus_address)
 
                 _LOGGER.info(
                     "%s@0x%02x component destroyed",
                     type(component).__name__,
-                    i2c_address,
+                    i2c_bus_address,
                 )
         else:
             _LOGGER.warning(
                 "%s@0x%02x component not found, unable to unload entity (pin %d).",
-                f"{DOMAIN}@0x{i2c_address:02x}",
-                i2c_address,
+                f"{DOMAIN}@0x{i2c_bus_address}",
+                i2c_bus_address,
```

Comment:
Not matching "%s@0x%02x..." formatting string above but I would probably remove this and change formatting string to "%s@..." ... not sure why I used/printed i2c_addresses twice/my bad